aviatrix/resource_aviatrix_site2cloud.go

```diff
@@ -439,16 +439,32 @@ func resourceAviatrixSite2CloudCreate(d *schema.ResourceData, meta interface{})
 		return fmt.Errorf("active_active_ha can't be enabled if HA isn't enabled for site2cloud connection")
 	}
 
+	if s2c.ConnType != "mapped" && s2c.ConnType != "unmapped" {
+		return fmt.Errorf("'connection_type' should be 'mapped' or 'unmapped'")
+	}
+
+	gateway := &goaviatrix.Gateway{
+		GwName: s2c.GwName,
+	}
+
+	gw, err := client.GetGateway(gateway)
```

Comment:
We should avoid calling more APIs here.

Reply:
this is a special case only for transit gateway, any suggestions about how to get the type of a gateway?

Reply:
have to call an API here to get the gateway type